pkg/controllers/user/alert/controller.go

```diff
@@ -0,0 +1,289 @@
+
+	stateSyncer := statesyncer.NewStateSyncer(cluster, alertmanager)
+	go stateSyncer.Run(ctx.Done())
+
+	sysWatcher := watcher.NewSysComponentWatcher(cluster, alertmanager)
+	go sysWatcher.Watch(ctx.Done())
+
+	podWatcher := watcher.NewPodWatcher(cluster, alertmanager)
+	go podWatcher.Watch(ctx.Done())
+
+	nodeWatcher := watcher.NewNodeWatcher(cluster, alertmanager)
+	go nodeWatcher.Watch(ctx.Done())
+
+	ssWatcher := watcher.NewStatefulsetWatcher(cluster, alertmanager)
+	go ssWatcher.Watch(ctx.Done())
+
+	depWatcher := watcher.NewDeploymentWatcher(cluster, alertmanager)
+	go depWatcher.Watch(ctx.Done())
+
+	dsWatcher := watcher.NewDaemonsetWatcher(cluster, alertmanager)
+	go dsWatcher.Watch(ctx.Done())
+
+	watcher.StartEventWatcher(cluster, alertmanager)
+
+	initClusterPreCanAlerts(clusterAlerts, cluster.ClusterName)
+
+}
+
+func initClusterPreCanAlerts(clusterAlerts v3.ClusterAlertInterface, clusterName string) {
+	etcdRule := &v3.ClusterAlert{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "clusteralert-etcd",
+		},
+		Spec: v3.ClusterAlertSpec{
+			ClusterName: clusterName,
+			AlertCommonSpec: v3.AlertCommonSpec{
+				DisplayName:           "Alert for etcd",
+				Description:           "Pre-can Alert for etcd component",
+				Severity:              "critical",
+				InitialWaitSeconds:    180,
+				RepeatIntervalSeconds: 3600,
+			},
+			TargetSystemService: v3.TargetSystemService{
+				Condition: "etcd",
+			},
+		},
+	}
+
+	if _, err := clusterAlerts.Create(etcdRule); err != nil {
+		logrus.Infof("Failed to create pre-can rules for etcd: %v", err)
+	}
+
+	cmRule := &v3.ClusterAlert{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "clusteralert-controllermanager",
+		},
+		Spec: v3.ClusterAlertSpec{
+			ClusterName: clusterName,
+			AlertCommonSpec: v3.AlertCommonSpec{
+				DisplayName:           "Alert for controller-manager",
+				Description:           "Pre-can Alert for controller-manager component",
+				Severity:              "critical",
+				InitialWaitSeconds:    180,
+				RepeatIntervalSeconds: 3600,
+			},
+			TargetSystemService: v3.TargetSystemService{
+				Condition: "controller-manager",
+			},
+		},
+	}
+
+	if _, err := clusterAlerts.Create(cmRule); err != nil {
+		logrus.Infof("Failed to create pre-can rules for controller manager: %v", err)
+	}
+
+	schedulerRule := &v3.ClusterAlert{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "clusteralert-scheduler",
+		},
+		Spec: v3.ClusterAlertSpec{
+			ClusterName: clusterName,
+			AlertCommonSpec: v3.AlertCommonSpec{
+				DisplayName:           "Alert for scheduler",
+				Description:           "Pre-can Alert for scheduler component",
+				Severity:              "critical",
+				InitialWaitSeconds:    180,
+				RepeatIntervalSeconds: 3600,
+			},
+			TargetSystemService: v3.TargetSystemService{
+				Condition: "scheduler",
+			},
+		},
+	}
+
+	if _, err := clusterAlerts.Create(schedulerRule); err != nil {
+		logrus.Infof("Failed to create pre-can rules for scheduler: %v", err)
```

Comment:
should be logged at least in Warn. Also do we want to proceed with the rest if pre-can rule creation fails?

Reply:
same applies to other places in this file where we just log the error in info

Reply:
I will use Warn. Yep, we should proceed even the pre-can rule creation fails